src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -161,21 +163,46 @@
-     * will result in a spliterator serving approximatively {@code N/2} elements (depending on whether N is even or not).
-     * As such, splitting is possible as long as {@code N >= 2}. The spliterator returns segments that feature the same
-     * scope as the given segment.
+     * The returned spliterator splits this segment according to the specified element layout; that is,
+     * if the supplied layout has size N, then calling {@link Spliterator#trySplit()} will result in a spliterator serving
+     * approximately {@code S/N/2} elements (depending on whether N is even or not), where {@code S} is the size of
+     * this segment. As such, splitting is possible as long as {@code S/N >= 2}. The spliterator returns segments that feature the same
+     * scope as this given segment.
      * <p>
      * The returned spliterator effectively allows to slice this segment into disjoint sub-segments, which can then
      * be processed in parallel by multiple threads.
      *
-     * @param layout the layout to be used for splitting.
+     * @param elementLayout the layout to be used for splitting.
      * @return the element spliterator for this segment
-     * @throws IllegalStateException if the segment is not <em>alive</em>, or if access occurs from a thread other than the
-     * thread owning this segment
+     * @throws IllegalArgumentException if this segment size is not a multiple of the size of {@code elementLayout}.
+     */
+    Spliterator<MemorySegment> spliterator(MemoryLayout elementLayout);
+
+    /**
+     * Returns a sequential {@code Stream} over disjoint slices (whose size matches that of the specified layout)
+     * in this segment. Calling this method is equivalent to the following code:
+     * <blockquote><pre>{@code
+    StreamSupport.stream(segment.spliterator(), false);
+     * }</pre></blockquote>
+     *
+     * @param elementLayout the layout to be used for splitting.
+     * @return a sequential {@code Stream} over disjoint slices in this segment.
+     * @throws IllegalArgumentException if this segment size is not a multiple of the size of {@code elementLayout}.
+     */
+    Stream<MemorySegment> stream(MemoryLayout elementLayout);
+
+    /**
+     * Returns a parallel {@code Stream} over disjoint slices (whose size matches that of the specified layout)
+     * in this segment. Calling this method is equivalent to the following code:
+     * <blockquote><pre>{@code
+    StreamSupport.stream(segment.spliterator(), true);
+     * }</pre></blockquote>
+     *
+     * @param elementLayout the layout to be used for splitting.
+     * @return a parallel {@code Stream} over disjoint slices in this segment.
+     * @throws IllegalArgumentException if this segment size is not a multiple of the size of {@code elementLayout}.
      */
-    Spliterator<MemorySegment> spliterator(SequenceLayout layout);
+    Stream<MemorySegment> parallelStream(MemoryLayout elementLayout);
```

Comment:
I don't think we need this method. Except for collections we generally don't use this pattern, and instead encourage the use of `Stream.parallel()` in such cases.